src/grafana_agent.py

```diff
@@ -275,11 +274,27 @@ def _update_status(self):
 
         Sets unit status to either Waiting or Active.
         """
-        if relations := self.model.relations.get("metrics-endpoint"):
-            if len(relations):
-                if not len(self.model.relations[REMOTE_WRITE_RELATION_NAME]):
-                    self.unit.status = WaitingStatus("no related Prometheus remote-write")
-                    return
+        # Make sure every incoming relation has a matching outgoing relation
+        for incoming, outgoing in [
+            # K8s
+            ("metrics-endpoint", "send-remote-write"),
+            ("logging-provider", "logging-consumer"),
+            ("grafana-dashboards-provider", "grafana-dashboards-consumer"),
+            # Machine
+            ("cos-agent", "send-remote-write"),
+            ("cos-agent", "logging-consumer"),
+            ("cos-agent", "grafana-dashboards-consumer"),
+        ]:
```

Comment:
`NamedTuple` would be _great_ here.

Reply:
Seems like a namedtuple would be a bit overly verbose in this very narrowly scoped use case:

```diff
-        for incoming, outgoing in [
-            ("metrics-endpoint", "send-remote-write"),
+        RelPair = namedtuple("RelPair", ["incoming", "outgoing"])
+        for pair in [
+            RelPair("metrics-endpoint", "send-remote-write"),
```
and then use `pair.incoming` and `pair.outgoing`...

Reply:
It is a hit to verbosity, which is why I only suggested a very light `NamedTuple` rather than something bigger, but the clarify is an improvement over raw tuples IMO, and it makes it easier to mock out/reason about in unit tests or similar.